pkg/controller/config/dns.go

```diff
@@ -40,17 +42,65 @@ func (h *configHandler) Handle(req router.Request, resp router.Response) error {
 		return err
 	}
 
+	domain, err := h.handleSecret(req, resp, cfg)
+	if err != nil {
+		return err
+	}
+
+	if err = h.handleIngress(req, resp, domain, cfg.AcornDNS, cfg.IngressClassName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// purgeRecordsIfDisabling checks if we are transitioning AcorDNS from an enabled state to a disabled state and if so calls the
+// acorn DNS service to purge all records for the domain. It is expected that string variable returned by this function
+// will be set as the labels.AcornDNSState annotation on the acorn-dns secret
+func purgeRecordsIfDisabling(req router.Request, domain string, cfg *apiv1.Config, dnsSecret *corev1.Secret,
```

Comment:
Just moved this function up unchanged.

Reply:
Purely from an ease-of-review standpoint, i'd personally shy away from just moving a function like this. just throw throws the diff off.